lib/src/util/dart_type_utilities.dart

```diff
@@ -40,6 +40,12 @@ class DartTypeUtilities {
         element.allSupertypes.any(predicate);
   }
 
+  static bool extendsClass(DartType type, String className, String library) =>
```

Comment:
I suspect that we want to allow the `type` to either extend or implement the given class, which means that this has to handle mixins and interfaces as well as superclasses.